prometheus/metric.go

```diff
@@ -184,8 +185,15 @@ func (m *withExemplarsMetric) Write(pb *dto.Metric) error {
 			if i < len(pb.Histogram.Bucket) {
 				pb.Histogram.Bucket[i].Exemplar = e
 			} else {
-				// This is not possible as last bucket is Inf.
-				panic("no bucket was found for given exemplar value")
+				// The +Inf bucket should be explicitly added if there is an exemplar for it, similar to non-const histogram logic in https://github.com/prometheus/client_golang/blob/main/prometheus/histogram.go#L357-L365.
+				b := &dto.Bucket{
+					CumulativeCount: proto.Uint64(pb.Histogram.Bucket[len(pb.Histogram.GetBucket())-1].GetCumulativeCount()),
+					UpperBound:      proto.Float64(math.Inf(1)),
+					Exemplar:        e,
+				}
+				pb.Histogram.Bucket = append(pb.Histogram.Bucket, b)
+				break
```

Comment:
Thanks for addressing this, but I think one thing is still not addressed: https://github.com/prometheus/client_golang/pull/1094#discussion_r931227718

Reply:
will fix that for you if you don't mind - I am preparing next release

Reply:
merging and will fix in separate PR.

Reply:
sure, also response in https://github.com/prometheus/client_golang/pull/1094#discussion_r931227718